crates/noirc_frontend/src/hir/def_collector/errors.rs

```diff
@@ -22,6 +26,8 @@ pub enum DefCollectorErrorKind {
     PathResolutionError(PathResolutionError),
     #[error("Non-struct type used in impl")]
     NonStructTypeInImpl { span: Span },
+    #[error("Feature not implemented")]
+    SimpleError { primary_message: String, secondary_message: String, span: Span },
```

Comment:
I used this simple error as generic error, because it was hard to know each error that could occur in advance. I think, converted all SimpleError to proper type can happen is separate PR.

Reply:
Creating an error kind for each error shouldn't be too much extra code. I'd like to avoid merging another general error type since we just had a bunch of PRs to remove all the general error types.